source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -222,6 +223,7 @@ Http::FilterHeadersStatus JsonTranscoderFilter::decodeHeaders(Http::HeaderMap& h
     // just pass-through the request to upstream.
     return Http::FilterHeadersStatus::Continue;
   }
+  has_http_body_output_ = hasHttpBodyAsOutputType();
```

Comment:
Since the behavior below depends on non-streaming handling, (i.e. buffer response until trailer), lets only set this when server_streaming is false.

Reply:
Hi @lizan sorry, I need to clarify this. Do you want to make sure that we only set `has_http_body_output_` to true when `server_streaming` is false and `hasHttpBodyAsOutputType()` is true?

Reply:
yes, `hasHttpBodyAsOutputType() && !server_streaming`

Reply:
Ah OK, done. Thanks.